numpy/core/records.py

```diff
@@ -423,6 +423,11 @@ def __new__(subtype, shape, dtype=None, buf=None, offset=0, strides=None,
                                       strides=strides, order=order)
         return self
 
+    def __array_finalize__(self, obj):
+        if type(obj) is not type(self):
+            # invoke __setattr__
+            self.dtype = self.dtype
```

Comment:
@ahaldane - In principle, at this point `self.dtype is obj.dtype` -- I think it would read better to just write `self.dtype = obj.dtype`, and expand your comment. e.g.,
```
    # invoke __setattr__, which for void dtypes will do
    # self.dtype = sb.dtype((record, obj.dtype))
    self.dtype = obj.dtype
```